create.go

```diff
@@ -0,0 +1,14 @@
+package statusresouce
+
+import (
+	"context"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+
+	// TODO add version if not tracked
```

Comment:
This should be straight forward. What we need to configure here is the appropriate rest client like we do in `operatorkit` for finalizers. Then the abstracted patch should just work. We might want to implement an interface on the provider CRs that allows us to fetch the tracked version bundle version. Alternatively we could also configure some getter to do so, which would be the specific key function we have available everywhere. Feedback welcome.

Reply:
REST client makes sense for me